renku/cli/dataset.py

```diff
@@ -248,14 +248,28 @@ def unlink(client, name, include, exclude, yes):
 @pass_local_client(clean=True, commit=True)
 def remove(client, names):
     """Delete a dataset."""
+    from renku.models.refs import LinkReference
     datasets = {client.dataset_path(name) for name in names}
+
     with progressbar(
-        datasets, item_show_func=lambda item: str(item) if item else ''
+        datasets,
+        label='Removing metadata files'.ljust(30),
+        item_show_func=lambda item: str(item) if item else '',
     ) as bar:
         for dataset in bar:
             if dataset and dataset.exists():
                 dataset.unlink()
-        click.secho('OK', fg='green')
+
+    with progressbar(
+        list(LinkReference.iter_items(client, common_path='datasets')),
+        label='Removing aliases'.ljust(30),
+        item_show_func=lambda item: item.name if item else '',
+    ) as bar:
+        for ref in bar:
+            if ref.reference in datasets:
+                ref.delete()
+
+    click.secho('OK', fg='green')
```

Comment:
Please don't output success if nothing happened. It's just confusing from the users perspective

Reply:
https://github.com/SwissDataScienceCenter/renku-python/pull/497/files#diff-5ee2736a38fb1ec55353bdd962e82066R258

Reply:
Consider the case where you have multiple datasets. Example:

```renku dataset
ID        NAME           CREATED              AUTHORS
--------  -------------  -------------------  ---------
0ad1cb9a  some-dataset2  2019-03-19 16:39:46
9436e36c  my-dataset     2019-02-28 16:48:09  sam
d40e0ce6  some-dataset   2019-03-09 16:34:34  sam
c53b021e  somedataset    2019-03-06 14:43:03  sam
```

And now as a user I make a mistake where I want to delete multiple datasets:

```renku dataset rm some-dataset2 some-dataset22                                                                  2 ↵
OK
```

In this case, we are reporting to the user that operation was successful and that we achieved what he wanted. Which is not the case cause we failed to remove the second dataset since it does not exist. We should not report OK in this case.

Let's create a ticket for this case to handle or handle it here.

Reply:
I've added an additional check.